src/zeek.bif

```diff
@@ -2961,26 +2961,32 @@ function bytestring_to_count%(s: string, is_le: bool &default=F%): count
 	static const bool host_bigendian = false;
 #endif
 	const u_char *p = s->Bytes();
-	unsigned int i;
+	unsigned int i = 0;
 
-	switch ( s->Len() ) {
-	case sizeof(uint8_t):
+	if ( s->Len() == 0 )
+		{
+		return zeek::val_mgr->Count(0);
+		}
+	if ( s->Len() == sizeof(uint8_t) )
 		{
 		uint8_t value = 0;
 		memcpy(&value, p, sizeof(uint8_t));
 		return zeek::val_mgr->Count(value);
 		}
-
-	case sizeof(uint16_t):
+	else if ( s->Len() <= sizeof(uint16_t) )
 		{
 		uint16_t value = 0;
 
 		if ( (host_bigendian && is_le) || (! host_bigendian && ! is_le) )
 			{
 			char buf[sizeof(uint16_t)];
+			memset(buf, 0, sizeof(uint16_t));
 			char *d = &buf[sizeof(uint16_t)-1];
 
-			for ( i = 0; i < sizeof(uint16_t); i++ )
+			if ( is_le )
```

Comment:
I'm struggling with these `is_le` checks here and below — is it intentional that here it's `if ( is_le )` and below it's `if ( ! is_le )`?

Reply:
See above about removing the changes on this block 😃